relentless/optimize/objective.py

```diff
@@ -279,9 +281,35 @@ def compute(self, directory=None):
             The result, which has unknown value ``None`` and known gradient.
 
         """
+        # run simulation and use result to compute gradient
         sim = self.simulation.run(self.target, self.potentials, directory, self.communicator)
         sim_ens = self.thermo.extract_ensemble(sim)
+        gradient = self.gradient(sim_ens)
+
+        # optionally write output to directory
+        if directory is not None and (self.communicator is None or self.communicator.rank == self.communicator.root):
+            for n,p in enumerate(self.potentials.pair.potentials):
+                p.save(directory.file('potential.{}.json'.format(n)))
```

Comment:
Looking at this, I am realizing that calling this "potential" is probably not the best idea, in case we add other types of potentials in future. Maybe we should rename the file to `pair_potential.*.json` or something like that?
```suggestion
                p.save(directory.file('pair_potential.{}.json'.format(n)))
```